pkg/logql/parser_test.go

```diff
@@ -745,6 +745,24 @@ func TestParse(t *testing.T) {
 				col:  13,
 			},
 		},
+		{
+			// missing "by" clause
+			in: `{foo="bar"} dedup (label)`,
```

Comment:
```suggestion
			in: `{foo="bar"} | dedup (label)`,
```

Ensure the pipe is correctly set so the error will be for the missing `by|without`

Reply:
Great point - that expresses the intent much more clearly